include/hello_world/version.hpp

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+/// The major version number
+#define VERSION_MAJOR 0
```

Comment:
I recommend doing `HELLO_VERSION_MAJOR` or `HELLOWORLD_VERSION_MAJOR` even though this verbose. The reason is that it indicates to a user forking hpp-skel, that they should replace it with `THEIRNAME_VERSION_MAJOR` and the reason that is important is because `#defines` are global to the program and could easily conflict if another header you don't control used `VERSION_MAJOR`. For example if two libraries, both based on hpp-skel forgot to update these `#define` names to be different then they would clash when both those headers were included and this would cause trouble (the value would not predictably be equal to what you think it should) and an annoying compiler warning would be issued confusing users compiling the program.

The rarer situation, but also one we want to avoid, by ensuring `#defines` are unique is if some other major library in the world chose to use `VERSION_MAJOR`. What if zlib used that? Then you could never include `hpp-skel` with zlib without this trouble.

In the future this will hopefully go away when we move to using "modules": https://clang.llvm.org/docs/Modules.html